test/test.js

```diff
@@ -113,6 +114,31 @@ afterAll(async({server, httpsServer}) => {
 });
 
 describe('Puppeteer', function() {
+  describe('Downloader', function() {
+    it('should download and extract linux binary', async({server}) => {
+      const downloadsFolder = await mkdtempAsync(TMP_FOLDER);
+      const downloader = puppeteer.createDownloader({
+        platform: 'linux',
+        path: downloadsFolder,
+        host: server.PREFIX
+      });
+      const revisionInfo = downloader.revisionInfo('123456');
+      server.setRoute(revisionInfo.url.substring(server.PREFIX.length), (req, res) => {
+        server.serveFile(req, res, '/chromium-linux.zip');
+      });
+
+      expect(downloader.platform()).toBe('linux');
+      expect(await downloader.canDownload('100000')).toBe(false);
+      expect(await downloader.canDownload('123456')).toBe(true);
+
+      await downloader.download('123456');
+      expect(await readFileAsync(revisionInfo.executablePath, 'utf8')).toBe('LINUX BINARY\n');
```

Comment:
the executablePath is defined before the binary is downloaded?

Reply:
yes, the revisionInfo could be called for both downloaded and non-downloaded revisions.